contrib/ros/envs/sawyer/reach_env.py

```diff
@@ -0,0 +1,123 @@
+"""
+Reach task for the sawyer robot
+"""
+import collections
+
+import gym
+import numpy as np
+
+from contrib.ros.envs.sawyer.sawyer_env import SawyerEnv
+from contrib.ros.robots.sawyer import Sawyer
+from contrib.ros.worlds.empty_world import EmptyWorld
+from garage.core import Serializable
+
+
+class ReachEnv(SawyerEnv, Serializable):
+    def __init__(self,
+                 initial_goal,
+                 initial_joint_pos,
+                 sparse_reward=False,
+                 simulated=False,
+                 distance_threshold=0.05,
+                 target_range=0.15,
+                 robot_control_mode='position'):
+        Serializable.quick_init(self, locals())
+
+        self._distance_threshold = distance_threshold
+        self._target_range = target_range
+        self._sparse_reward = sparse_reward
+        self.initial_goal = initial_goal
+        self.goal = self.initial_goal.copy()
+
+        self._sawyer = Sawyer(
+            initial_joint_pos=initial_joint_pos,
+            control_mode=robot_control_mode)
+        self._world = EmptyWorld(simulated)
+
+        SawyerEnv.__init__(
+            self, simulated=simulated, robot=self._sawyer, world=self._world)
```

Comment:
doesn't SawyerEnv set `self._robot` and `self._world`?

Reply:
oh right!